2023/08/wtchangdm/day8.rs

```diff
@@ -0,0 +1,190 @@
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Copy)]
+enum Direction {
+    Left,
+    Right,
+}
+
+impl Direction {
+    fn from(ch: &char) -> Self {
+        match ch {
+            'L' => Self::Left,
+            'R' => Self::Right,
+            _ => panic!("come on it's impossible"),
+        }
+    }
+}
+
+struct Map {
+    cursor: usize,
+    directions: Vec<Direction>,
+    nodes: HashMap<String, [String; 2]>,
+}
+
+impl Map {
+    fn from(input: &[String]) -> Self {
+        let mut nodes: HashMap<String, [String; 2]> = HashMap::new();
```

Comment:
Unless you are intended to make it explicit, there is no need to write the type out here.

```suggestion
        let mut nodes = HashMap::new();
```

Reply:
Yeah, actually I use just trying to express the left and right direction but too lazy to bother with dedicated struct, hence the weird type here.